src/libopensc/card-muscle.c

```diff
@@ -483,6 +483,9 @@ static int muscle_init(sc_card_t *card)
 	if(card->type == SC_CARD_TYPE_MUSCLE_JCOP241) {
 		card->caps |= SC_CARD_CAP_APDU_EXT;
 	}
+	if(card->type == SC_CARD_TYPE_MUSCLE_JCOP242_NO_APDU_EXT) {
+	        /* JCOP v2.4.2 card that doesn't support extended APDUs */
```

Comment:
I have a JCOP 2.4.2 that actually *has* support for extended length APDUs.

As far as I can see, the Tyfone card does not customize the ATR. Do you have an other possibility do differentiate between your card and other 2.4.2 cards? If so, use this proberty to distinguish extended/short length support.